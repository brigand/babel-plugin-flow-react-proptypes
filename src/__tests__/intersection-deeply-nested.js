const babel = require('babel-core');

const content = `
var React = require('react');

import type {NamedType, OtherNamedType, ThirdNamedType} from "./types";

class D extends React.Component {
   props: NamedType & {
    foo: number,
    bar: NamedType,
    baz: NamedType & OtherNamedType & {
      foo: "number", bar: NamedType & OtherNamedType
    } & ThirdNamedType,
  };
};

export default D;
`;

it('intersection-exported-type-alias-unexported-object-literal', () => {
  const res = babel.transform(content, {
    babelrc: false,
    presets: ['es2015', 'stage-1', 'react'],
    plugins: ['syntax-flow', require('../')],
  }).code;
  expect(res).toMatchSnapshot();
});
