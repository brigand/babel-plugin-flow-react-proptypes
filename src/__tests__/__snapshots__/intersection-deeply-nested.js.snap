exports[`test intersection-exported-type-alias-unexported-object-literal 1`] = `
"\"use strict\";

Object.defineProperty(exports, \"__esModule\", {
  value: true
});

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError(\"Cannot call a class as a function\"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError(\"this hasn\'t been initialised - super() hasn\'t been called\"); } return call && (typeof call === \"object\" || typeof call === \"function\") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== \"function\" && superClass !== null) { throw new TypeError(\"Super expression must either be null or a function, not \" + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

var React = require(\'react\');

var babelPluginFlowReactPropTypes_proptype_ThirdNamedType = require(\"./types\").babelPluginFlowReactPropTypes_proptype_ThirdNamedType || require(\"prop-types\").any;

var babelPluginFlowReactPropTypes_proptype_OtherNamedType = require(\"./types\").babelPluginFlowReactPropTypes_proptype_OtherNamedType || require(\"prop-types\").any;

var babelPluginFlowReactPropTypes_proptype_NamedType = require(\"./types\").babelPluginFlowReactPropTypes_proptype_NamedType || require(\"prop-types\").any;

var D = function (_React$Component) {
  _inherits(D, _React$Component);

  function D() {
    _classCallCheck(this, D);

    return _possibleConstructorReturn(this, (D.__proto__ || Object.getPrototypeOf(D)).apply(this, arguments));
  }

  return D;
}(React.Component);

D.propTypes = Object.assign({}, babelPluginFlowReactPropTypes_proptype_NamedType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_NamedType, {
  foo: require(\"prop-types\").number.isRequired,
  bar: typeof babelPluginFlowReactPropTypes_proptype_NamedType === \"function\" ? babelPluginFlowReactPropTypes_proptype_NamedType : require(\"prop-types\").shape(babelPluginFlowReactPropTypes_proptype_NamedType),
  baz: require(\"prop-types\").shape(Object.assign({}, babelPluginFlowReactPropTypes_proptype_NamedType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_NamedType, babelPluginFlowReactPropTypes_proptype_OtherNamedType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_OtherNamedType, {
    foo: require(\"prop-types\").oneOf([\"number\"]).isRequired,
    bar: require(\"prop-types\").shape(Object.assign({}, babelPluginFlowReactPropTypes_proptype_NamedType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_NamedType, babelPluginFlowReactPropTypes_proptype_OtherNamedType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_OtherNamedType)).isRequired
  }, babelPluginFlowReactPropTypes_proptype_ThirdNamedType === require(\"prop-types\").any ? {} : babelPluginFlowReactPropTypes_proptype_ThirdNamedType)).isRequired
});
;

exports.default = D;"
`;
